components/org.wso2.carbon.identity.api.server.action.management/org.wso2.carbon.identity.api.server.action.management.v1/src/main/java/org/wso2/carbon/identity/api/server/action/management/v1/core/ServerActionManagementService.java

```diff
@@ -295,10 +315,17 @@ private Action buildUpdatingAction(ActionUpdateModel actionUpdateModel) {
                     .build();
         }
 
+        ActionRule actionRule = null;
+        if (actionUpdateModel.getRule() != null) {
+            actionRule = RuleMapper.toActionRule(actionUpdateModel.getRule(), actionType,
```

Comment:
Shall we move the `actionUpdateModel.getRule() != null` check inside the `toActionRule` method, so we do not need to check it whenever we use that method.

Reply:
Well the responsibility of the RuleMapper is to transform the rule. And rule being optional is a part of the flow.
So, IMO, RuleMapper should not check for this null check and it should expect that a rule instance is passed.